client/engine/engine.go

```diff
@@ -65,20 +76,30 @@ func New(msg messageservice.MessageService, chain chainservice.ChainService, sto
 	return e
 }
 
+func (e *Engine) ToApi() <-chan ObjectiveChangeEvent {
+	return e.toApi
+}
+
 // Run kicks of an infinite loop that waits for communications on the supplied channels, and handles them accordingly
 func (e *Engine) Run() {
 	for {
+		var res ObjectiveChangeEvent
 		select {
 		case apiEvent := <-e.FromAPI:
-			e.handleAPIEvent(apiEvent)
+			res = e.handleAPIEvent(apiEvent)
```

Comment:
The other option is that I could pass in the `e.toApi` channel to `handleFoo` and it could just send events directly. However it seemed nice to isolate dealing with the channel in one place.